sstables/compaction_manager.cc

```diff
@@ -256,12 +256,15 @@ class user_initiated_backlog_tracker final : public compaction_backlog_tracker::
     virtual void remove_sstable(sstables::shared_sstable sst)  override { }
 };
 
-future<> compaction_manager::submit_major_compaction(column_family* cf) {
-    if (_stopped) {
-        return make_ready_future<>();
-    }
+lw_shared_ptr<compaction_manager::task>
+compaction_manager::submit_major_compaction(column_family* cf) {
     auto task = make_lw_shared<compaction_manager::task>();
     task->compacting_cf = cf;
+    if (_stopped) {
+        task->compaction_done = make_ready_future<>();
```

Comment:
I wonder if that's really what we want to return to the user?
But that's out of the scope of this change.

Reply:
Also compaction_done seems to be initialized to a (shared) ready future.
So I'm not sure how waiting for it works (even without this change).

Reply:
@raphaelsc ?